python/tskit/trees.py

```diff
@@ -3713,22 +3719,37 @@ def haplotypes(self, impute_missing_data=False, missing_data_character="-"):
 
         :return: An iterator over the haplotype strings for the samples in
             this tree sequence.
-        :param bool impute_missing_data: If True, the allele assigned to any
-            isolated samples is the ancestral state; that is, we impute
-            missing data as the ancestral state. Default: False.
+        :param bool isolated_as_missing: If True, the allele assigned to
+            missing (isolated) samples is ``missing_data_character``. If False,
+            missing (isolated) samples will be assigned the ancestral state or
+            that of the mutation above them, if any.
+            Default: True.
```

Comment:
how about
```
        :param bool isolated_as_missing: If True, the allele assigned to
            missing samples (i.e., isolated samples without mutations) is the ``missing_data_character``. If False,
            missing samples will be assigned the ancestral state.
            Default: True.
```

Reply:
I guess I'm interpreting our position to be that samples with a mutation are *not* missing - do you agree, @hyanwong, @jeromekelleher?

Reply:
Yes, completely

Reply:
Done

Reply:
Yes, we've fixed that in one of the in-flight PRs.